akka-stream/src/main/scala/akka/stream/impl/io/ByteStringParser.scala

```diff
@@ -78,6 +111,8 @@ private[akka] abstract class ByteStringParser[T] extends GraphStage[FlowShape[By
  */
 private[akka] object ByteStringParser {
 
+  val CompactionThreshold = 16
```

Comment:
I propose an alternative solution as this is a common issue: https://github.com/akka/akka/issues/21865

I will fix it separately as the fix in this PR is important.